prov/verbs/src/verbs_info.c

```diff
@@ -680,8 +681,9 @@ static int fi_ibv_alloc_info(struct ibv_context *ctx, struct fi_info **info,
 		goto err;
 	}
 
-	snprintf(fi->domain_attr->name, name_len + 1, "%s%s",
-		 ctx->device->name, ep_dom->suffix);
+	if (snprintf(fi->domain_attr->name, name_len + 1, "%s%s",
+		 ctx->device->name, ep_dom->suffix) >= name_len + 1)
+		 fi->domain_attr->name[name_len] = '\0';
 	fi->domain_attr->name[name_len] = '\0';
```

Comment:
I think it would be better to use `calloc` instead of `malloc` for `fi_domain_attr::name` above. So, we can drop this line

Reply:
That will not fix the problem.

snprintf will do the following

char a[4];
...
snprintf(a, 4, "%s", "12345");

Will result in "a" being:

a = "1234" _without_ any '\0' bytes.

So calloc would not be a fix here.

Reply:
if you overflow with snprintf all the zero'ed elements of calloc will be over run and there still will not be a trailing '\0'.  So I don't think that would help.

Reply:
Ah I see what you are saying...  calloc(1, name_len+1) then use name_len leaves a trailing null no matter what...  I did not see the allocation...  I think we can drop this patch.

Reply:
> calloc(1, name_len+1) then use name_len leaves a trailing null no matter what.

Yes, that's what I meant